Classes/Package.php

```diff
@@ -21,23 +23,52 @@ public function boot(Bootstrap $bootstrap)
         $dispatcher = $bootstrap->getSignalSlotDispatcher();
 
         $dispatcher->connect(Node::class, 'nodePropertyChanged', function (NodeInterface $node, $propertyName, $_, $newValue) use ($bootstrap) {
-            if ($propertyName !== 'identifier' || empty($newValue) || !$node->getNodeType()->isOfType('Neos.Form.Builder:IdentifierMixin')) {
+            if ($propertyName !== 'identifier' || empty($newValue) || !$node->getNodeType()->isOfType(self::NODE_TYPE_IDENTIFIER_MIXIN)) {
                 return;
             }
 
-            /** @noinspection PhpUndefinedMethodInspection */
-            $flowQuery = (new FlowQuery([$node]))->context(['invisibleContentShown' => true, 'removedContentShown' => true, 'inaccessibleContentShown' => true]);
-            $possibleIdentifier = $initialIdentifier = $newValue;
-            $i = 1;
-            /** @noinspection PhpUndefinedMethodInspection */
-            while ($flowQuery
-                    ->closest('[instanceof Neos.Form.Builder:NodeBasedForm]')
-                    // [identifier=".."] matches the Form Element identifier, [_identiier!="..."] excludes the current node
-                    ->find(sprintf('[instanceof Neos.Form.Builder:IdentifierMixin][identifier="%s"][_identifier!="%s"]', $possibleIdentifier, $node->getIdentifier()))
-                    ->count() > 0) {
-                $possibleIdentifier = $initialIdentifier . '-' . $i++;
+            $this->setUniqueFormElementIdentifier($node, $newValue);
+        });
+
+        $dispatcher->connect(Node::class, 'nodeAdded', function (NodeInterface $node) use ($bootstrap) {
+            try {
+                $identifier = $node->getProperty('identifier');
+
+                if (empty($identifier) || !$node->getNodeType()->isOfType(self::NODE_TYPE_IDENTIFIER_MIXIN)) {
+                    return;
+                }
+            } catch (\Neos\ContentRepository\Exception\NodeException $e) {
+                return;
             }
-            $node->setProperty('identifier', $possibleIdentifier);
+
+            $this->setUniqueFormElementIdentifier($node, $identifier);
         });
     }
+
+    /**
+     * @param NodeInterface $node
+     * @param string $identifier
+     * @throws \Neos\Eel\Exception
+     */
+    protected function setUniqueFormElementIdentifier(NodeInterface $node, string $identifier): void
```

Comment:
This is not meant to be extensible, is it?
```suggestion
    private function setUniqueFormElementIdentifier(NodeInterface $node, string $identifier): void
```